probe-rs/src/flashing/flasher.rs

```diff
@@ -326,21 +326,40 @@ impl Flasher {
     ) -> Result<(), FlashError> {
         progress.started_filling();
 
-        let result = self.run_verify(session, progress, |active| {
+        fn fill_pages(
+            layout: &mut FlashLayout,
+            progress: &FlashProgress,
+            mut read: impl FnMut(u64, &mut [u8]) -> Result<(), FlashError>,
+        ) -> Result<(), FlashError> {
             for fill in layout.fills.iter() {
                 let t = Instant::now();
                 let page = &mut layout.pages[fill.page_index()];
-
                 let page_offset = (fill.address() - page.address()) as usize;
                 let page_slice = &mut page.data_mut()[page_offset..][..fill.size() as usize];
 
-                active.read_flash(fill.address(), page_slice)?;
+                read(fill.address(), page_slice)?;
 
                 progress.page_filled(fill.size(), t.elapsed());
             }
 
             Ok(())
-        });
+        }
+
+        let result = if self.flash_algorithm.pc_read.is_some() {
+            self.run_verify(session, progress, |active| {
+                fill_pages(layout, progress, |address, data| {
+                    active.read_flash(address, data)
+                })
+            })
+        } else {
+            // Initialize, then deinitialize the flash loader. This may execute code necessary to
+            // enable the flash chip for reading, for example on the RP2040.
+            self.run_verify(session, progress, |_| Ok(()))?;
```

Comment:
This seems very weird to me. I believe it fixes the issue but is there no more intuitive way?

Reply:
I don't think so, code needs to run to attach external flash. Maybe we could coerce the reset debug sequence to let the bootloader run, but that smells fragile.

Reply:
FWIW the proper solution would be to implement the Read function but I don't expect anyone to do that.

Reply:
I will have a look :)